tests/include/test/macros.h

```diff
@@ -166,14 +166,16 @@
  * \param size2     Size of the second buffer in bytes.
  *                  This expression may be evaluated multiple times.
  */
-#define ASSERT_COMPARE(p1, size1, p2, size2)                          \
-    do                                                                  \
-    {                                                                   \
+#define TEST_BUFFERS_EQUAL(p1, size1, p2, size2)               \
```

Comment:
I don't like `BUFFERS_EQUAL` because it's ambiguous: it could mean “are these the same buffers?” rather than “do these buffers have the same content?”. In fact my default interpretation for “are the buffers equal?” is “are these the same buffers?”, although I would suspect that this is in fact comparing contents because there's not much of a need to test for buffer equality whereas there's a frequent need to test for buffer content equality.

`memcmp` is a well-known name in C, so I think `TEST_MEMORY_COMPARE` would be a better name.